src/modules/75-cf/utils/UsageLimitUtils.ts

```diff
@@ -0,0 +1,80 @@
+/*
+ * Copyright 2022 Harness Inc. All rights reserved.
+ * Use of this source code is governed by the PolyForm Shield 1.0.0 license
+ * that can be found in the licenses directory at the root of this repository, also available at
+ * https://polyformproject.org/wp-content/uploads/2020/06/PolyForm-Shield-1.0.0.txt.
+ */
+
+import { BannerType } from '@common/layouts/Constants'
+import { formatToCompactNumber } from '@cf/utils/CFUtils'
+import type { UseStringsReturn } from 'framework/strings'
+
+interface BannerText {
+  message: string
+  bannerType: BannerType
+}
+
+export const getBannerText = (
+  getString: UseStringsReturn['getString'],
+  additionalLicenseProps: Record<string, any>,
+  count?: number,
+  limit?: number
+): BannerText => {
+  const { isEnterpriseEdition, isFreeEdition, isTeamEdition } = additionalLicenseProps
+
+  const clientMauUsageCount = Number(count)
+  const clientMauPlanLimit = Number(limit)
```

Comment:
Since `count` and `limit` are optional, passing them through `Number` could result in `NaN` and throw off the rest of the function

Reply:
it wont throw off the rest of the function as its caught by `clientMauPlanLimit > 0`. `NaN > 0` will return `false`

Reply:
`count` & `limit` need to be converted to a number as (I think) the types are `any`. Can you suggest a better way to do this?

Reply:
The values are used in a couple of calculations before that as well. Instead of letting it do those, you could return early, or detect early in the caller and just not call this function at all (thereby letting you set them as required).
```suggestion
  const clientMauUsageCount = Number(count)
  const clientMauPlanLimit = Number(limit)
  
  if (isNaN(clientMauUsageCount) || isNaN(clientMauPlanLimit)) {
    // return early
  }
```

If if's coz they are typed as `any`, but always are genuinely numbers, then a type assertion is a better option.